### PR TITLE
feat: ability to define mood parameters

### PR DESCRIPTION
**Pull request by djaiss**
This PR lets you define the parameters that users will be able to use to track their moods.

Mood tracking parameters are defined per vault.

We provide some default values for the moods, but users can modify it.

In another PR, we'll let users actually track their moods.

image

### CODE REVIEW

**Comment by djaiss** [comment]
Then, mood tracking should be 

```
Schema::create('mood_tracking', function (Blueprint $table) {
            $table->id();
            $table->unsignedBigInteger('contact_id');
            $table->unsignedBigInteger('mood_tracking_parameter_id');
            $table->datetime('rated_at');
            $table->text('note')->nullable();
            $table->timestamps();
            $table->foreign('mood_tracking_parameter_id')->references('id')->on('mood_tracking_parameters')->onDelete('cascade');
        });
```

File: app/Domains/Vault/ManageVault/Services/CreateVault.php [hunk 1]
```diff
@@ -113,4 +115,47 @@ private function populateDefaultContactImportantDateTypes(): void
...
         ]);
     }
+
+    private function populateMoodTrackingParameters(): void
+    {
+        (new CreateMoodTrackingParameter())->execute([
+            'account_id' => $this->data['account_id'],
+            'author_id' => $this->author->id,
+            'vault_id' => $this->vault->id,
+            'label' => trans('vault.settings_mood_tracking_parameters_awesome'),
```
Comment (asbiin) [inline app/Domains/Vault/ManageVault/Services/CreateVault.php]: Translated label should not go into the database directly...

File: app/Domains/Vault/ManageVaultSettings/Services/CreateMoodTrackingParameter.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        $this->validateRules($data);
+
+        // determine the new position
+        $newPosition = $this->vault->moodTrackingParameters()
+            ->max('position');
+        $newPosition++;
+
+        $this->moodTrackingParameter = MoodTrackingParameter::create([
+            'vault_id' => $data['vault_id'],
+            'label' => $data['label'],
```
Comment (asbiin) [inline app/Domains/Vault/ManageVaultSettings/Services/CreateMoodTrackingParameter.php:60]: Translated label should not go into the database directly...

File: database/migrations/2023_01_07_005110_create_mood_tracking_parameters_table.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            $table->integer('position')->nullable();
+            $table->timestamps();
+            $table->foreign('vault_id')->references('id')->on('vaults')->onDelete('cascade');
+        });
+
+        Vault::chunk(200, function ($vaults) {
+            foreach ($vaults as $vault) {
+                DB::table('mood_tracking_parameters')->insert([
+                    'vault_id' => $vault->id,
+                    'label' => trans('vault.settings_mood_tracking_parameters_awesome'),
```
Comment (asbiin) [inline database/migrations/2023_01_07_005110_create_mood_tracking_parameters_table.php]: Translated label should not go into the database directly...

**Review by asbiin — CHANGES_REQUESTED** [review]
OK, but I did not tested it on dark mode, what the behavior there?

File: app/Domains/Vault/ManageVault/Services/CreateVault.php (hunk 1 above)
Comment (djaiss) [reply app/Domains/Vault/ManageVault/Services/CreateVault.php]: fixed

File: database/migrations/2023_01_07_005110_create_mood_tracking_parameters_table.php (hunk 3 above)
Comment (djaiss) [reply database/migrations/2023_01_07_005110_create_mood_tracking_parameters_table.php]: fixed

File: app/Domains/Vault/ManageVaultSettings/Services/CreateMoodTrackingParameter.php (hunk 2 above)
Comment (djaiss) [reply app/Domains/Vault/ManageVaultSettings/Services/CreateMoodTrackingParameter.php:60]: fixed

**Pull request merged**